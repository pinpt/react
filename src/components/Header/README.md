### Displays the title of your site, the description, and a subscribe button, and can contain search and your logo (coming soon)

[Full Docs](https://react.preview.pinpoint.com/?path=/docs/components-header) | [Github](https://github.com/pinpt/react/tree/master/src/components/Header)

#### Theming

The parent className for styling this component is `.Header`

<details>
	<summary>Base Theme Styles</summary>

```css
.headerWrapper {
	background-color: var(--header-bg-color);
}

.Header {
	@apply flex py-5 flex-col;
}

.Header .top {
	@apply flex justify-between;
}

.Header .right {
	@apply flex gap-x-4;
}

.Header .center {
	@apply py-8 flex flex-col content-center text-center;
}

.Header .title {
	@apply font-semibold text-4xl md:text-5xl py-6 transition-colors duration-200;
	color: var(--header-title-color);
}

.Header .description {
	color: var(--header-description-color);
}

.Header .action {
	@apply mt-6 mx-auto;
}

.Header .Subscribe {
	color: var(--header-subscribe-color);
}
```

</details>
