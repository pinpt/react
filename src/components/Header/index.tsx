import { ReactElement } from 'react';

import type { IActionLinkProps } from '../Internal/ActionLink';

import type { ISearchBarProps } from '../Search/Bar';
import type { ISubscribeProps } from '../Subscribe';
import type { IThemeToggleProps } from '../ThemeToggle';
import type { ILogoProps } from '../Logo';

export interface IHeaderProps {
	className?: string;
	logo?: ReactElement<ILogoProps | IActionLinkProps>;
	title?: string;
	description?: string;
	subscribe?: ReactElement<ISubscribeProps>;
	themeToggle?: ReactElement<IThemeToggleProps>;
	search?: ReactElement<ISearchBarProps>;
	href?: string;
	onClick?: () => void;
}

const Header = (props: IHeaderProps) => {
	const { className = '', title, description, subscribe, themeToggle, search, logo, href, onClick } = props;

	return (
		<div className={`Pinpoint Header ${className}`}>
			<div className="top">
				<div className="left">{logo}</div>
				<div className="right">
					{search}
					{themeToggle}
				</div>
			</div>
			<div className="center">
				{title && (
					<a
						href={href}
						title={title}
						onClick={onClick}
						className={(href || onClick) ? 'link' : undefined}
					>
						<h1 className="title">{title}</h1>
					</a>
				)}
				{description && <div className="description">{description}</div>}
				{subscribe && <div className="action">{subscribe}</div>}
			</div>
		</div>
	);
};

export default Header;
