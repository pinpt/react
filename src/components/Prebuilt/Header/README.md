# Prebuilt Header

The prebuilt components are the easiest to use since they work right out of the box, but provide super simple customization options. As its simplest, a prebuilt component can be rendered as such:

```jsx
import { Prebuilt, useSite } from '@pinpt/react';

const { Header } = Prebuilt;

const MySiteHeader = () => {
	const site = useSite();

	return <Header site={site} entries={entries} />;
};
```

That's it! You'll get a fully functioning header rendered with your data. Every part of the prebuilt component can be overridden, and if you want to hide a part of the app, simply return an empty fragment from the renderer:

```jsx
import { Prebuilt, useSite } from '@pinpt/react';

const { Header } = Prebuilt;

const MySiteHeader = () => {
	const site = useSite();

	return <Header site={site} renderLogo={() => <img src="path/to/my/awesome/logo" />} renderThemeToggler={() => <></>} />;
};
```
