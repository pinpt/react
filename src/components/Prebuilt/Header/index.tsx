import { ReactElement } from 'react';
import { ISite } from '../../../lib';
import BaseHeader from '../../Header';
import Logo, { ILogoProps } from '../../Logo';
import Search from '../../Search';
import { ISearchBarProps } from '../../Search/Bar';
import Subscribe, { ISubscribeProps } from '../../Subscribe';
import ThemeToggle, { IThemeToggleProps } from '../../ThemeToggle';

export interface IPrebuiltHeaderProps {
	className?: string;
	searchTerm?: string;
	site: ISite;
	handleSearch?: (value: string) => void;
	handleSelectHome?: () => void;
	renderLogo?: (site: ISite) => ReactElement<ILogoProps>;
	renderSearch?: (site: ISite) => ReactElement<ISearchBarProps>;
	renderSubscribe?: (site: ISite) => ReactElement<ISubscribeProps>;
	renderThemeToggle?: (site: ISite) => ReactElement<IThemeToggleProps>;
	title?: string;
	description?: string;
	onToggleMenu?: () => void;
	mobileMenu?: boolean;
}

const Header = (props: IPrebuiltHeaderProps) => {
	const {
		className = '',
		searchTerm,
		site,
		handleSearch,
		handleSelectHome,
		renderLogo,
		renderSearch,
		renderSubscribe,
		renderThemeToggle,
		title,
		description,
		onToggleMenu,
		mobileMenu,
	} = props;

	return (
		<BaseHeader
			className={`Prebuilt ${className}`}
			title={title ?? site.theme?.title ?? site.name}
			description={description ?? site.theme?.description ?? site.name}
			onClick={() => handleSelectHome?.()}
			subscribe={renderSubscribe?.(site) ?? <Subscribe className="Prebuilt" href="/subscription/subscribe" />}
			themeToggle={renderThemeToggle?.(site) ?? <ThemeToggle className="Prebuilt" />}
			search={
				renderSearch?.(site) ?? (
					<Search.Bar defaultValue={searchTerm} onSubmit={handleSearch} className="Prebuilt" />
				)
			}
			onToggleMenu={onToggleMenu}
			mobileMenu={mobileMenu}
			logo={
				renderLogo?.(site) ?? (
					<Logo
						src={site.logoUrl}
						className="Prebuilt"
						href={site.theme?.logoLink ?? site.url}
						title="Header Logo Home Page"
					/>
				)
			}
		/>
	);
};

export default Header;
