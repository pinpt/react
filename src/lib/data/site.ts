import type { Site } from '../types/site';
import { executeAPI } from '../fetch';

export const fetchSite = async (siteId: string): Promise<Site> => {
	const { site } = await executeAPI(`/site-api/v1/site/${siteId}`);
	return site;
};

type AnalyticsResult = Record<string, { claps: number; pageviews: number }>;

export const fetchAnalytics = async (siteId: string): Promise<AnalyticsResult> => {
	const { data } = await executeAPI(`/site-api/v1/site/${siteId}/analytics`);
	return data;
};
