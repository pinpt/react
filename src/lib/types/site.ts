export interface ISite {
	id: string;
	logoUrl?: string;
	mailingAddress?: string;
	url: string; // the canonical url for the site
	slug: string;
	name: string;
	private: boolean; // true if private and should not be indexable by search engines
	theme?: {
		title?: string;
		social?: Record<string, string>;
		description?: string;
		logoLink?: string;
		copyright?: string;
		eulaLink?: string;
		privacyLink?: string;
		tosLink?: string;
		homepage?: {
			tags?: string[];
		};
	};
	scriptPlugins?: Record<string, any>;
	basePath: string;
}
