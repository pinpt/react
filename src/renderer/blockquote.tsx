import { registerNode, recurseIntoChildren, NodeProps } from './register';

const BlockQuote = ({ node }: NodeProps) => {
	return <blockquote>{recurseIntoChildren(node)}</blockquote>;
};

registerNode('blockquote', (node) => <BlockQuote node={node} />);
