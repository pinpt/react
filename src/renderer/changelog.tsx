import { ReactYouTubeLite as Youtube } from 'react-youtube-lite';
import { getFileUrl } from '../lib/file_metadata';
import { slugifyContent } from '../lib/string';
import { ChangelogCoverMedia, ChangelogMediaType } from '../lib/types';
import { Content as Doc } from './';

const ImageMedia = ({ src, title = '' }: { src: string; title?: string }) => {
	return (
		<div className="image">
			<img src={src} alt={title} />
		</div>
	);
};

const VideoMedia = ({ src }: { src: string }) => {
	return (
		<div className="video">
			<video controls src={src} />
		</div>
	);
};

const YoutubeMediaSSR = ({ id, poster = 'hqdefault' }: { id: string; poster?: string }) => {
	// the component isn't SSR friendly so this is basically used to sort of mimic it for generated code
	const posterUrl = `https://i.ytimg.com/vi/${id}/${poster}.jpg`;
	return (
		<>
			{poster && <link rel="preload" href={posterUrl} as="image" />}
			<div
				className="youtube ryt-lite embed-responsive aspect-ratio-16/9"
				style={{ backgroundImage: `url(${posterUrl})` }}
			>
				<div className="lty-playbtn" data-url={`https://www.youtube-nocookie.com/embed/${id}?autoplay=1`} />
			</div>
		</>
	);
};

const YoutubeMedia = ({
	id,
	metadata,
	staticMode,
}: {
	id: string;
	metadata?: Record<string, any>;
	staticMode?: true;
}) => {
	return (
		<div className="youtube">
			{!staticMode ? (
				<Youtube url={`https://www.youtube.com/watch?v=${id}`} poster={metadata?.poster} />
			) : (
				<YoutubeMediaSSR id={id} poster={metadata?.poster} />
			)}
		</div>
	);
};

const Media = ({ media, title, staticMode }: { media?: ChangelogCoverMedia; title?: string; staticMode?: true }) => {
	if (!media) {
		return <></>;
	}
	let content: React.ReactElement;
	switch (media.type) {
		case ChangelogMediaType.None: {
			return <></>;
		}
		case ChangelogMediaType.Image: {
			content = <ImageMedia src={getFileUrl(media.value)} title={title} />;
			break;
		}
		case ChangelogMediaType.Video: {
			content = <VideoMedia src={getFileUrl(media.value)} />;
			break;
		}
		case ChangelogMediaType.Youtube: {
			content = <YoutubeMedia id={media.value} metadata={media.metadata} staticMode={staticMode} />;
			break;
		}
	}
	return <div className="media-container">{content}</div>;
};

interface ChangelogProps {
	id: string;
	title?: string;
	content?: any;
	coverMedia?: ChangelogCoverMedia;
	limit?: number;
	divider?: boolean;
	staticMode?: true;
}

const ChangelogLog = (props: ChangelogProps) => {
	return (
		<article className="changelog notebook-editor read-only">
			{props.coverMedia && (
				<section className="covermedia">
					<Media media={props.coverMedia} title={props.title} staticMode={props.staticMode} />
				</section>
			)}
			<section className="ProseMirror">
				<Doc node={props.content} limit={props.limit} />
				{props.limit && props.content.content?.length > props.limit && (
					<div className="continue">
						<a href={slugifyContent(props.id, props.title)}>Continue Reading â†’</a>
					</div>
				)}
			</section>
			{props.divider && <hr className="divider" />}
		</article>
	);
};

export default ChangelogLog;
