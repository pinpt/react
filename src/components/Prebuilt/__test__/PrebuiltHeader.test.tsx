import renderer from 'react-test-renderer';
import Header from '../Header';
import site from '../__data__/testSite.json';

test('Test default state', () => {
	const component = renderer.create(<Header site={site} />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test custom className', () => {
	const component = renderer.create(<Header className="test-custom" site={site} />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test custom basepath', () => {
	const component = renderer.create(<Header className="test-custom" site={{ ...site, basePath: '/blog' }} />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
