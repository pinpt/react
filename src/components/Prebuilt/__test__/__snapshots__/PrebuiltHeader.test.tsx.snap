// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test custom basepath 1`] = `
<div
  className="Pinpoint Header Prebuilt test-custom"
>
  <div
    className="top"
  >
    <div
      className="left"
    >
      <a
        aria-label="Header Logo Home Page"
        className="Pinpoint Internal Link Pinpoint Logo sm Prebuilt"
        href="https://pinpoint.com"
        title="Header Logo Home Page"
      >
        <img
          alt=""
          className="Pinpoint Image image"
          decoding="async"
          height={320}
          loading="lazy"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw=="
          width={320}
        />
      </a>
    </div>
    <div
      className="right"
    >
      <div
        className="Pinpoint Search Bar Prebuilt"
      >
        <svg
          aria-hidden="true"
          className="icon"
          data-icon="search"
          data-prefix="fas"
          focusable="false"
          height={16}
          role="img"
          viewBox="0 0 512 512"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            fill="currentColor"
          />
        </svg>
        <form
          className="form"
          onSubmit={[Function]}
        >
          <input
            className="input"
            onChange={[Function]}
            placeholder="Search"
            value=""
          />
        </form>
      </div>
      <div
        className="Pinpoint ThemeToggle Prebuilt"
      >
        <button
          aria-label="Toggle Theme"
          className="mode"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="center"
  >
    <a
      className="link"
      onClick={[Function]}
      title="Pinpoint Changelog"
    >
      <h1
        className="title"
      >
        Pinpoint Changelog
      </h1>
    </a>
    <div
      className="description"
    >
      New updates and improvements to Pinpoint
    </div>
    <div
      className="action"
    >
      <a
        className="Pinpoint Internal Link Pinpoint Subscribe transition-opacity opacity-0 Prebuilt"
        href="https://changelog.pinpoint.com/blog/subscription/subscribe"
      >
        <svg
          aria-hidden="true"
          className="icon"
          data-icon="envelope"
          data-prefix="fas"
          focusable="false"
          height={14}
          role="img"
          viewBox="0 0 512 512"
          width={14}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
            fill="currentColor"
          />
        </svg>
        <span
          className="text"
        >
          Subscribe to Updates
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`Test custom className 1`] = `
<div
  className="Pinpoint Header Prebuilt test-custom"
>
  <div
    className="top"
  >
    <div
      className="left"
    >
      <a
        aria-label="Header Logo Home Page"
        className="Pinpoint Internal Link Pinpoint Logo sm Prebuilt"
        href="https://pinpoint.com"
        title="Header Logo Home Page"
      >
        <img
          alt=""
          className="Pinpoint Image image"
          decoding="async"
          height={320}
          loading="lazy"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw=="
          width={320}
        />
      </a>
    </div>
    <div
      className="right"
    >
      <div
        className="Pinpoint Search Bar Prebuilt"
      >
        <svg
          aria-hidden="true"
          className="icon"
          data-icon="search"
          data-prefix="fas"
          focusable="false"
          height={16}
          role="img"
          viewBox="0 0 512 512"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            fill="currentColor"
          />
        </svg>
        <form
          className="form"
          onSubmit={[Function]}
        >
          <input
            className="input"
            onChange={[Function]}
            placeholder="Search"
            value=""
          />
        </form>
      </div>
      <div
        className="Pinpoint ThemeToggle Prebuilt"
      >
        <button
          aria-label="Toggle Theme"
          className="mode"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="center"
  >
    <a
      className="link"
      onClick={[Function]}
      title="Pinpoint Changelog"
    >
      <h1
        className="title"
      >
        Pinpoint Changelog
      </h1>
    </a>
    <div
      className="description"
    >
      New updates and improvements to Pinpoint
    </div>
    <div
      className="action"
    >
      <a
        className="Pinpoint Internal Link Pinpoint Subscribe transition-opacity opacity-100 Prebuilt"
        href="https://changelog.pinpoint.com/subscription/subscribe"
      >
        <svg
          aria-hidden="true"
          className="icon"
          data-icon="envelope"
          data-prefix="fas"
          focusable="false"
          height={14}
          role="img"
          viewBox="0 0 512 512"
          width={14}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
            fill="currentColor"
          />
        </svg>
        <span
          className="text"
        >
          Subscribe to Updates
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`Test default state 1`] = `
<div
  className="Pinpoint Header Prebuilt "
>
  <div
    className="top"
  >
    <div
      className="left"
    >
      <a
        aria-label="Header Logo Home Page"
        className="Pinpoint Internal Link Pinpoint Logo sm Prebuilt"
        href="https://pinpoint.com"
        title="Header Logo Home Page"
      >
        <img
          alt=""
          className="Pinpoint Image image"
          decoding="async"
          height={320}
          loading="lazy"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw=="
          width={320}
        />
      </a>
    </div>
    <div
      className="right"
    >
      <div
        className="Pinpoint Search Bar Prebuilt"
      >
        <svg
          aria-hidden="true"
          className="icon"
          data-icon="search"
          data-prefix="fas"
          focusable="false"
          height={16}
          role="img"
          viewBox="0 0 512 512"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            fill="currentColor"
          />
        </svg>
        <form
          className="form"
          onSubmit={[Function]}
        >
          <input
            className="input"
            onChange={[Function]}
            placeholder="Search"
            value=""
          />
        </form>
      </div>
      <div
        className="Pinpoint ThemeToggle Prebuilt"
      >
        <button
          aria-label="Toggle Theme"
          className="mode"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="center"
  >
    <a
      className="link"
      onClick={[Function]}
      title="Pinpoint Changelog"
    >
      <h1
        className="title"
      >
        Pinpoint Changelog
      </h1>
    </a>
    <div
      className="description"
    >
      New updates and improvements to Pinpoint
    </div>
    <div
      className="action"
    >
      <a
        className="Pinpoint Internal Link Pinpoint Subscribe transition-opacity opacity-0 Prebuilt"
        href="https://changelog.pinpoint.com/subscription/subscribe"
      >
        <svg
          aria-hidden="true"
          className="icon"
          data-icon="envelope"
          data-prefix="fas"
          focusable="false"
          height={14}
          role="img"
          viewBox="0 0 512 512"
          width={14}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
            fill="currentColor"
          />
        </svg>
        <span
          className="text"
        >
          Subscribe to Updates
        </span>
      </a>
    </div>
  </div>
</div>
`;
